src/reducer.js

```diff
@@ -142,16 +152,30 @@ const RegistrationLiteReducer = (state = DEFAULT_STATE, action) => {
         case PAY_RESERVATION: {
             return { ...state, checkout: payload.response, reservation: null, userProfile: null, invitation: null };
         }
-        case GET_MY_INVITATION:{
-            return {...state, invitation: payload.response};
+        case GET_MY_INVITATION: {
+            return { ...state, invitation: payload.response };
         }
-        case CLEAR_MY_INVITATION:{
-            return {...state, invitation: null};
+        case CLEAR_MY_INVITATION: {
+            return { ...state, invitation: null };
         }
         case UPDATE_CLOCK: {
             const { timestamp } = payload;
             return { ...state, nowUtc: timestamp };
         }
+        case REQUEST_TICKET_DISCOUNT: {
+            const { promoCode } = payload;
+            return { ...state, promoCode: { ...state.promoCode, code: promoCode, loading: true } }
+        }
+        case REMOVE_TICKET_DISCOUNT: {
+            return { ...state, promoCode: { loading: true } }
+        }
+        case GET_TICKET_DISCOUNT: {
+            const hasDiscount = payload.response.data.some(e => e.hasOwnProperty('cost_with_applied_discount'))
+            const ticket_types = hasDiscount ? payload.response.data : state.ticketTypes;
+            const code = hasDiscount ? state.promoCode.code : '';
+            const promoCode = { code, hasDiscount, loading: false }
+            return { ...state, promoCode, ticketTypes: ticket_types, promoCode }
```

Comment:
![image](https://github.com/fntechgit/summit-registration-lite/assets/2577122/ad42bee6-5b59-4c9c-b91a-12755fc0c4fe)
this is a dupe declaration and its marked by the linter